explorer/src/components/common/TableSettings.tsx

```diff
@@ -0,0 +1,305 @@
+import { Bars3Icon, FunnelIcon, MagnifyingGlassIcon, PencilIcon } from '@heroicons/react/24/outline'
+import React, { useState } from 'react'
+import { AvailableColumn, FilterOption } from 'types/table'
+
+interface TableSettingsProps {
+  tableName: string
+  totalLabel: string
+  availableColumns: AvailableColumn[]
+  selectedColumns: string[]
+  filterOptions: FilterOption[]
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  filters: Record<string, any>
+  showTableSettings: string | null
+  addExtraIcons?: React.ReactNode
+  showSettings: (setting: 'columns' | 'filters' | 'search') => void
+  hideSettings: () => void
+  handleColumnChange: (column: string, checked: boolean) => void
+  handleFilterChange: (filterName: string, value: string | boolean) => void
+}
+
+export const TableSettings: React.FC<TableSettingsProps> = ({
+  tableName,
+  totalLabel,
+  availableColumns,
+  selectedColumns,
+  filterOptions,
+  filters,
+  showTableSettings,
+  addExtraIcons,
+  showSettings,
+  hideSettings,
+  handleColumnChange,
+  handleFilterChange,
+}) => {
+  const [mobileMenuOpen, setMobileMenuOpen] = useState(false)
+
+  return (
+    <div className='mb-4 w-full' id='accordion-open' data-accordion='open'>
+      <h2 id='accordion-open-heading-1'>
+        <div className='flex w-full items-center justify-between truncate pb-5 text-left font-light text-gray-900 dark:text-white/75'>
+          <span className='flex items-center'>
+            {tableName} ({totalLabel})
+          </span>
+          <div className='flex items-center'>
+            <div className='hidden sm:flex'>
+              {addExtraIcons && addExtraIcons}
+              <MagnifyingGlassIcon
+                className='m-4 size-10 rounded-full border-2 border-white p-1'
+                stroke='#FFFFFF'
+                key='search'
+                onClick={() =>
+                  showTableSettings !== 'search' ? showSettings('search') : hideSettings()
+                }
+              />
+              <PencilIcon
+                className='m-4 size-10 rounded-full border-2 border-white p-1'
+                stroke='#FFFFFF'
+                key='pencil'
+                onClick={() =>
+                  showTableSettings !== 'columns' ? showSettings('columns') : hideSettings()
+                }
+              />
+              <FunnelIcon
+                className='m-4 size-10 rounded-full border-2 border-white p-1'
+                stroke='#FFFFFF'
+                key='funnel'
+                onClick={() =>
+                  showTableSettings !== 'filters' ? showSettings('filters') : hideSettings()
+                }
+              />
```

Comment:
In light view the Icons are very difficult to see. 
![image](https://github.com/user-attachments/assets/bf6ed394-c5b9-4914-9947-8fe704d8d164)

Reply:
This as been fixed